src/pages/loginPage.ts

```diff
@@ -0,0 +1,61 @@
+import {Page} from '@playwright/test';
+import PlaywrightWrapper from '../helper/wrapper/playwrightWrappers';
+import basePage from './basePage';
+
+export default class LoginPage {
+  private base: PlaywrightWrapper;
+
+  constructor(private page: Page) {
+    this.base = new PlaywrightWrapper(page);
+  }
+
+  logInBtn = this.page.locator("//a/span[text()='Log in']");
+  emailInput = this.page.locator("//input[@name='email']");
+  passwordInput = this.page.locator("//input[@name='password']");
+  submitBtn = this.page.locator("//button[@type='submit']");
+
+  async enterUsernameAndPassowrd(userRole: string) {
+    let email: string;
+    let password: string;
+
+    switch (userRole) {
+      case 'admin':
+        email = process.env.ADMIN_EMAIL;
+        password = process.env.ADMIN_PASSWORD;
+        break;
+      case 'publisher':
+        email = process.env.PUBLISHER_EMAIL;
+        password = process.env.PUBLISHER_PASSWORD;
+        break;
+      case 'advertiser':
+          email = process.env.ADVERTISER_EMAIL;
+          password = process.env.ADVERTISER_PASSWORD;
+          break;
+      case 'agency':
+          email = process.env.AGENCY_EMAIL;
+          password = process.env.AGENCY_PASSWORD;
+          break;
+      default:
+        throw new Error(`Unsupported user role: ${userRole}`);
+    }
```

Comment:
I'd recommend moving this part of the code to separate function and make this implementation even simpler.